### PR TITLE
Upgrade to Spotlight 2

### PR DESCRIPTION
**Pull request by mejackreed**
Remaining items:

 - [x] update `spotlight_initializer.rb` to use `Spotlight::UploadFieldConfig`

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -11,48 +11,60 @@
...
+    # this has to come before the Blacklight + Spotlight routes to avoid getting routed as
+    # a document request.
+    resources :exhibits, path: '/', only: [] do
+      get "catalog/range_limit" => "spotlight/catalog#range_limit"
+    end
 
-  mount Blacklight::Engine => '/'
-  resources :solr_documents, only: [:show], path: '/catalog', controller: 'catalog'
+    concern :searchable, Blacklight::Routes::Searchable.new
+    # mount Blacklight::Engine => '/'
```
Comment (cbeer) [inline config/routes.rb]: Remove these commented out lines?

**Comment by cbeer** [comment]
For our first release, do we want to limit `i18n_locales` to just languages we know we care about or expect to have full translations for?

File: config/routes.rb (hunk 1 above)
Comment (mejackreed) [reply config/routes.rb]: 👍 done.

**Comment by mejackreed** [comment]
I could go either way on that. I'm kind of fine with people using the feature and finding new things with it. Do you see some other issue that might arise?

**Comment by jkeck** [comment]
I'm in the mind that we roll w/ what Blacklight provides (which is what I believe we've done here)

**Pull request merged**